Plugin/CsrfValidatorSkip.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Vindi\Payment\Plugin;
+
+use Magento\Framework\App\ProductMetadata;
+
+class CsrfValidatorSkip
+{
+
+    public function __construct(
+        ProductMetadata $productMetadata
+    ) {
+        $this->productMetadata = $productMetadata;
+    }
+
+    /**
+     * @param \Magento\Framework\App\Request\CsrfValidator $subject
+     * @param \Closure $proceed
+     * @param \Magento\Framework\App\RequestInterface $request
+     * @param \Magento\Framework\App\ActionInterface $action
+     */
+    public function aroundValidate(
+        $subject,
+        \Closure $proceed,
+        $request,
+        $action
+    ) {
+        if ($this->productMetadata->getVersion() < '2.3.0') {
+            $proceed($request, $action);
```

Comment:
Qual o motivo da implementação dessa classe?

Reply:
Esta classe permite que webhooks são recebidos pelo Magento corretamente, pois da versão 2.3.0 em diante, requisições de fora são barrados pelo Magento.

Reply:
Não entendi muito bem.
Se for menor que a versão 2.3.0 ele irá executar o **proceed**, caso contrário irá apenas retornar?
Esse é o correto? :thinking:

Reply:
Isso, assim webhooks não são barrados.